File: e6/python/solution1.py
import sys
class Solution:
    #@param A and B: sorted integer array A and B.
    #@return: A new sorted integer array
    def mergeSortedArray(self, A, B):
        # write your code here
        ret = []
        sza = len(A)
        szb = len(B)
        pa = 0
        pb = 0
        for i in range(sza + szb):
            if pa == sza:
                ret.append(B[pb])
                pb += 1
            elif pb == szb:
                ret.append(A[pa])
                pa += 1
            else:
                if A[pa] < B[pb]:
                    ret.append(A[pa])
                    pa += 1
                else:
                    ret.append(B[pb])
                    pb += 1
        return ret

def main(argv=None):
    if argv is None:
        argv = sys.argv
    lista = range(1, 10000000)
    #lista = range(0, 10)
    listb = [1,3,5]
    Solution().mergeSortedArray(lista, listb)


if __name__ == '__main__':
    main()

File: e6/java/Solution1.java
class Solution1 {
    /**
     * @param A and B: sorted integer array A and B.
     * @return: A new sorted integer array
     */
    public int[] mergeSortedArray(int[] A, int[] B) {
        // Write your code here
		int sza = A.length;
		int szb = B.length;
		int pa = 0;
		int pb = 0;
		int[] ret = new int[sza+szb];
		for (int i=0; i < sza+szb; i++) {
			if (pa >= sza) {
				ret[i] = B[pb];
				pb++;
			}
			else if (pb >= szb) {
				ret[i] = A[pa];
				pa++;
			}
			else {
				if (A[pa] < B[pb]) {
					ret[i] = A[pa];
					pa++;
				}
				else {
					ret[i] = B[pb];
					pb++;
				}
			}
		}
		return ret;
    }

	public static void main(String[] args) {
		int[] a = {1,3,5};
		int[] b = {1,2,3,4,5};
		int[] newArray = new Solution1().mergeSortedArray(a, b);
		for (int i:newArray) {
			System.out.println(i);
		}
	}
}
